compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -1451,8 +1451,10 @@ public void visit(BLangTypeDefinition typeDefinition) {
         definedType.flags |= typeDefSymbol.flags;
 
         typeDefinition.symbol = typeDefSymbol;
-
-        if (!typeDefinition.hasCyclicReference) {
+        if (typeDefinition.hasCyclicReference) {
+            // Workaround for https://github.com/ballerina-platform/ballerina-lang/issues/29742
+            typeDefinition.type.tsymbol = typeDefSymbol;
```

Comment:
Strange but calling `createLabelSymbol` we have created a copy of the type symbol. I am setting the copied symbol back to the type.